doc/source/developer_reference/writing_plugins.rst

```diff
@@ -39,6 +39,15 @@ proceeding onto guidance for specific plugin types.
 The Context
 ^^^^^^^^^^^
 
+N.B. In the below text, an iteration refers to a single job run. If only one
+iteration is specified for a job in the agenda, then a job is equivalent to
+an iteration. Each iteration will have a single folder. The number of folders
+will simply be equal to (sections * jobs * iterations/job), in addition to
+between 0 and an unpredictable number of additional folders for failed jobs.
+WA has a run hierarchy, with sections at the top and jobs below this.
+Iterations can be defined at any level (job-specific, or run-wide). All jobs
+are run in each of the section configurations.
+
```

Comment:
I'm not sure this belongs here all together. In order to better clarify what an `iteration` is would it make more sense to add it as a definition to the glossary and reference it when it is mentioned? 
And rather than trying to explain about the folder structure here maybe add a reference to the "Output Directory Structure" section at relevant part and make any additional clarifications there? 

(Also I think the term `job` is used slightly incorrect here, a job is an individual instance of a workload to be executed for example 1 workload entry in an agenda with 3 iterations will result in 3 jobs being generated.)

Reply:
In addition to the above,

- "In the below text,"  -- "In the following text," or simply "Below,"
- Change "folder" --> "directory"
- Please use the `.. note::` restructuredText admonition, rather than "N.B.".